main.go

```diff
@@ -2,57 +2,30 @@ package main
 
 import (
 	"bufio"
-	"errors"
 	"fmt"
 	"io"
 	"os"
 	"regexp"
-	"strconv"
 	"strings"
 
+	"github.com/bitrise-io/go-steputils/stepconf"
 	"github.com/bitrise-io/go-utils/command"
 	"github.com/bitrise-io/go-utils/fileutil"
 	"github.com/bitrise-io/go-utils/log"
-	"github.com/bitrise-tools/go-steputils/input"
 )
 
-// ConfigsModel ...
-type ConfigsModel struct {
-	BuildGradlePth    string
-	NewVersionName    string
-	NewVersionCode    string
-	VersionCodeOffset string
-}
-
-func createConfigsModelFromEnvs() ConfigsModel {
-	return ConfigsModel{
-		BuildGradlePth:    os.Getenv("build_gradle_path"),
-		NewVersionName:    os.Getenv("new_version_name"),
-		NewVersionCode:    os.Getenv("new_version_code"),
-		VersionCodeOffset: os.Getenv("version_code_offset"),
-	}
-}
-
-func (configs ConfigsModel) print() {
-	log.Infof("Configs:")
-	log.Printf("- BuildGradlePth: %s", configs.BuildGradlePth)
-	log.Printf("- NewVersionName: %s", configs.NewVersionName)
-	log.Printf("- NewVersionCode: %s", configs.NewVersionCode)
-	log.Printf("- VersionCodeOffset: %s", configs.VersionCodeOffset)
-}
-
-func (configs ConfigsModel) validate() error {
-	if err := input.ValidateIfPathExists(configs.BuildGradlePth); err != nil {
-		return errors.New("issue with input BuildGradlePth: " + err.Error())
-	}
-
-	if err := input.ValidateIfNotEmpty(configs.NewVersionCode); err != nil {
-		if err := input.ValidateIfNotEmpty(configs.NewVersionName); err != nil {
-			return errors.New("neither NewVersionCode nor NewVersionName are provided however, one of them is required")
-		}
-	}
+const (
+	// versionCode — A positive integer [...] -> https://developer.android.com/studio/publish/versioning
+	versionCodeRegexPattern = `^versionCode(?:\s|=)*(.*?)(?:\s|\/\/|$)`
+	// versionName — A string used as the version number shown to users [...] -> https://developer.android.com/studio/publish/versioning
+	versionNameRegexPattern = `^versionName(?:=|'|"|\s)+(.*?)(?:'|"|\s|\/\/|$)`
+)
 
-	return nil
+type config struct {
+	BuildGradlePth    string  `env:"build_gradle_path,file"`
+	NewVersionName    *string `env:"new_version_name"`
```

Comment:
Why not use just string and check if empty?

Reply:
we have this new feature in stepconf, and also it is a marshaler way of checking unset

Reply:
I am aware, just unsure if it is useful for string type as we discussed previously.